test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -197,7 +197,7 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, machine *
 	// Make sure bootstrap data is available and populated.
 	if machine.Spec.Bootstrap.DataSecretName == nil {
 		log.Info("Waiting for the Bootstrap provider controller to set bootstrap data")
-		conditions.MarkFalse(dockerMachine, infrav1.ContainerProvisionedCondition, infrav1.WaitingForBootstrapDataReason, clusterv1.ConditionSeverityInfo, "")
+		conditions.MarkFalse(dockerMachine, infrav1.ContainerProvisionedCondition, clusterv1.WaitingForControlPlaneAvailableReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
I'm not sure this should be changed, because there still could be cases where a docker machine is waiting for a data secret.
Instead, IMO we should distinguish when we are waiting for the control plane to be initialized. Something like

```
if !cluster.status.ControlPlaneInitialized {
    conditions.MarkFalse(dockerMachine, infrav1. WaitingForControlPlaneAvailableReason, infrav1.WaitingForBootstrapDataReason, clusterv1.ConditionSeverityInfo, "")
    return ctrl.Result{}, nil
}

if machine.Spec.Bootstrap.DataSecretName == nil {
    conditions.MarkFalse(dockerMachine, infrav1.ContainerProvisionedCondition, infrav1.WaitingForBootstrapDataReason, clusterv1.ConditionSeverityInfo, "")
    return ctrl.Result{}, nil
}
```